src/streaming-client.ts

```diff
@@ -0,0 +1,97 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import StreamingResponse from './models/streaming/StreamingResponse';
+
+export class RevAiStreamingClient extends events.EventEmitter {
+    accessToken: string;
+    config: AudioConfig;
+    baseUrl: string;
+    requests: PassThrough;
+    responses: PassThrough;
+    client: sockets.WebSocketClient;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        super();
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
+        this.requests = new PassThrough();
+        this.responses = new PassThrough({ objectMode: true });
+        this.client = new sockets.client();
+    }
+
+    start (): Duplex {
+        var self = this;
+        this.client.on('connectFailed', function(error: Error) {
+            self.emit('connectFailed', error);
+            self.requests.end();
+        });
+        this.client.on('connect', function(connection: any) {
+            self.emit('connect');
```

Comment:
shouldn't we return the `ConnectedStreamingResponse` here? It contains `JobId` which I imagine is crucial for users

Reply:
also these messages should be the past tense, so like `self.emit('connected')` and `self.emit('closed')` down below etc

Reply:
That response is also read from the responses output stream currently. I agree that it would be a better user experience if this returned it instead.

The standard for event names in javascript seems to be present tense. I can't find it written anywhere but based off the events emitted by streams and websockets